src/libraries/System.Private.CoreLib/src/System/Globalization/CompareInfo.OSX.cs

```diff
@@ -32,6 +37,67 @@ private unsafe int CompareStringNative(ReadOnlySpan<char> string1, ReadOnlySpan<
+            Debug.Assert(!target.IsEmpty);
+            Debug.Assert(!GlobalizationMode.UseNls);
+            Debug.Assert(target.Length != 0);
+
+            if (_isAsciiEqualityOrdinal && CanUseAsciiOrdinalForOptions(options))
+            {
+                if ((options & CompareOptions.IgnoreCase) != 0)
+                    return IndexOfOrdinalIgnoreCaseHelper(source, target, options, matchLengthPtr, fromBeginning);
+                else
+                    return IndexOfOrdinalHelper(source, target, options, matchLengthPtr, fromBeginning);
+            }
+            else
+            {
+                // GetReference may return nullptr if the input span is defaulted. The native layer handles
+                // this appropriately; no workaround is needed on the managed side.
+
+                fixed (char* pSource = &MemoryMarshal.GetReference(source))
+                fixed (char* pTarget = &MemoryMarshal.GetReference(target))
+                {
+                    NSRange result = Interop.Globalization.IndexOfNative(m_name, m_name.Length, pTarget, target.Length, pSource, source.Length, options, fromBeginning);
+                    if (matchLengthPtr == null)
+                        matchLengthPtr = &result.Length;
+                    *matchLengthPtr = result.Length;
+                    return result.Location;
+                }
+            }
+        }
+
+        private unsafe bool NativeStartsWith(ReadOnlySpan<char> source, ReadOnlySpan<char> prefix, CompareOptions options)
+        {
+            Debug.Assert(!GlobalizationMode.Invariant);
+            Debug.Assert(!GlobalizationMode.UseNls);
+
+            Debug.Assert(!prefix.IsEmpty);
+            Debug.Assert((options & (CompareOptions.Ordinal | CompareOptions.OrdinalIgnoreCase)) == 0);
+
+            fixed (char* pSource = &MemoryMarshal.GetReference(source)) // could be null (or otherwise unable to be dereferenced)
+            fixed (char* pPrefix = &MemoryMarshal.GetReference(prefix))
+            {
+                return Interop.Globalization.StartsWithNative(m_name, m_name.Length, pPrefix, prefix.Length, pSource, source.Length, options);
+            }
+        }
+
+        private unsafe bool NativeEndsWith(ReadOnlySpan<char> source, ReadOnlySpan<char> suffix, CompareOptions options)
+        {
+            Debug.Assert(!GlobalizationMode.Invariant);
+            Debug.Assert(!GlobalizationMode.UseNls);
+
+            Debug.Assert(!suffix.IsEmpty);
+            Debug.Assert((options & (CompareOptions.Ordinal | CompareOptions.OrdinalIgnoreCase)) == 0);
+
+            fixed (char* pSource = &MemoryMarshal.GetReference(source)) // could be null (or otherwise unable to be dereferenced)
+            fixed (char* pSuffix = &MemoryMarshal.GetReference(suffix))
+            {
+                return Interop.Globalization.EndsWithNative(m_name, m_name.Length, pSuffix, suffix.Length, pSource, source.Length, options);
+            }
+        }
```

Comment:
How come we aren't using 
```
            if (_isAsciiEqualityOrdinal && CanUseAsciiOrdinalForOptions(options))
            {
                if ((options & CompareOptions.IgnoreCase) != 0)
                    return StartsWithOrdinalIgnoreCaseHelper(source, prefix, options, matchLengthPtr);
                else
                    return StartsWithOrdinalHelper(source, prefix, options, matchLengthPtr);
            }
```
in these functions?

Because these are also similar to the ICU counterpart, could we just rely on that and have a guard around the interop function called?
So instead of having a `NativeEndsWith`, put a 
```
#if TARGET_OSX || TARGET_MACCATALYST || TARGET_IOS || TARGET_TVOS
    if (GlobalizationMode.Hybrid)
        return Interop.Globalization.StartsWithNative(...);
#endif
```

Reply:
As `IsPrefix` , `IsSuffix` (correspondingly `StartsWith`, `EndsWith`) with current implementation can't return matched length information that is why we are not using this part for `Hybrid` mode on `OSX`.